### DISCUSSION TITLE
Version 7.46.0-next.0

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#10792] [General]
## `disabled` prop

- new `disabled` props for `useForm`
- allow user to disable the entire form with all associated inputs

```tsx
useForm({
  disabled: true
})

const App = () => {
  const [disabled, setDisabled] = useState(false);
  const { register, handleSubmit, control } = useForm({
    disabled,
  });

  return (
    <form
      onSubmit={handleSubmit(async () => {
        setDisabled(true);
        await sleep(100);
        setDisabled(false);
      })}
    >
      <input
        type={'checkbox'}
        {...register('checkbox')}
        data-testid={'checkbox'}
      />
      <input type={'radio'} {...register('radio')} data-testid={'radio'} />
      <input type={'range'} {...register('range')} data-testid={'range'} />
      <select {...register('select')} data-testid={'select'} />
      <textarea {...register('textarea')} data-testid={'textarea'} />

      <Controller control={control} render={({ field }) => <input disabled={field.disabled}  />} name="test" />

      <button>Submit</button>
    </form>
  );
}
```

<hr /><em>This discussion was created from the release <a href='https://github.com/react-hook-form/react-hook-form/releases/tag/v7.46.0-next.0'>Version 7.46.0-next.0</a>.</em>

**Comment by nfrankeln** [comment]
Hi I'm new to web development could someone explain the implications of this? Why would one want to disable a form on submission. Is it simply to prevent multiple submissions?

**Reply by sajidkhanpk** [reply]
Yes, you want your form to be disabled while submitting to prevent any other interaction.

**Reply by bluebill1049** [reply]
That's correct. Image when you press enter and submit a long form.

**Comment by sajidkhanpk** [comment]
@bluebill1049 I mostly use MUI nowadays. So, does it work for Controller?

**Reply by bluebill1049** [reply]
there would be a disabled prop pass down for you to wire up with MUI input.

**Reply by sajidkhanpk** [reply]
Great!

**Comment by titenis** [comment]
Doesnt seem to work when disabled state is changed outside the form. For example if `disabled: isLoading` from react query, the form stays disabled even if `isLoading` becomes `false`. I cannot create sandbox, because cant select this version in codesandbox

**Reply by Moshyfawn** [reply]
I've managed to reproduce the issue you're describing. It seems to be related to [the `values` API](https://react-hook-form.com/docs/useform#values) usage. Omitting `values` seem to "fix" the problem.

_Side note: you can always init a project on your local machine and import it into codesandbox. [Here's a reproduction sandbox](https://codesandbox.io/p/github/Moshyfawn/rhf-disabled-test/draft/cranky-browser?file=%2Fsrc%2FApp.tsx%3A32%2C18)_

**Comment by Wekios** [comment]
What is the status of this? This is still not deployed or exposed or by the lib.

**Reply by bluebill1049** [reply]
next minor update
